modules/crypt.cpp

```diff
@@ -200,6 +342,41 @@ class CCryptMod : public CModule {
     }
 
     EModRet OnPrivNotice(CNick& Nick, CString& sMessage) override {
+        CString sCommand = sMessage.Token(0);
+        CString sOtherPubKey = sMessage.Token(1);
+
+        if ((sCommand.Equals("DH1080_INIT") || sCommand.Equals("DH1080_INIT_CBC")) && !sOtherPubKey.empty()) {
+            CString sSecretKey;
+            CString sTail = sMessage.Token(2); /* For fish10 */
+
+            /* remove trailing A */
+            if (sOtherPubKey.TrimSuffix("A") && DH1080_gen() && DH1080_comp(sOtherPubKey, sSecretKey)) {
+                PutModule("Received DH1080 public key from " + Nick.GetNick() + ", sending mine...");
+                PutIRC("NOTICE " + Nick.GetNick() + " :DH1080_FINISH " + sPubKey + "A" + (sTail.empty()?"":(" " + sTail)));
+                SetNV(Nick.GetNick().AsLower(), sSecretKey);
+                PutModule("Key for " + Nick.GetNick() + " successfully set.");
+                return HALT;
+            }
+            PutModule("Error in " + sCommand + " with " + Nick.GetNick() + ": " + (sSecretKey.empty()?"no secret key computed":sSecretKey));
+            return CONTINUE;
+
+        } else if (sCommand.Equals("DH1080_FINISH") && !sOtherPubKey.empty()) {
+            /*
+             * In theory we could get a DH1080_FINISH without us having sent a DH1080_INIT first,
+             * but then to have any use for the other user, they'd already have our pub key
+             */
+            CString sSecretKey;
+
+            /* remove trailing A */
+            if (sOtherPubKey.TrimSuffix("A") && DH1080_gen() && DH1080_comp(sOtherPubKey, sSecretKey)) {
```

Comment:
The weird trailing A is... weird. So, this stuff is Base64 encoded, but trailing = are dropped and afterwards trialing A are dropped?